rocketc/rocketc.py

```diff
@@ -538,3 +596,72 @@ def _get_list_groups(self, user_id, auth_token, query=""):
             groups = groups["groups"]
             for group in groups:
                 yield group["name"]
+
+    def _get_user_messages(self, group_name, latest="", oldest="", count=100):
+        """
+        Gets the messages from a user's private group.
+        """
+        api = self.api_rocket_chat
+        rocket_chat_group = api.search_rocket_chat_group(group_name)
+
+        if not rocket_chat_group['success']:
+            return None
+
+        group_id = rocket_chat_group['group']['_id']
+        messages = api.get_groups_history(room_id=group_id, latest=latest,
+                                          oldest=oldest, count=count)
+
+        if not messages["success"]:
+            return None
+
+        return [message for message in messages["messages"]
+                if message["u"]["username"] == self.user_data["username"]]
+
+    def _filter_by_reaction_and_user_role(self, messages, reaction):
+        """
+        Returns generator with filtered messages by a given reaction
+        """
+        for message in messages:
+            if not "reactions" in message:
+                continue
+            if not reaction in message["reactions"]:
+                continue
+            usernames = message["reactions"][reaction]["usernames"]
+
+            for username in usernames:
+                if self._validate_user_role(username):
+                    yield message
+                    break
+
+    def _validate_user_role(self, username):
+        """
+        Returns True if the user is privileged in teams discussions for
+        this course.
+        """
+        from openedx_dependencies import has_discussion_privileges, CourseStaffRole  # pylint: disable=relative-import
+
+        user = User.objects.get(username=username)
+
+        if user.is_staff:
+            return True
+        if CourseStaffRole(self.user_data["course_id"]).has_user(user):
+            return True
+        if has_discussion_privileges(user, self.user_data["course_id"]):
+            return True
+        return False
+
+    def _grading_discussions(self):
+        """
+        This method allows to grade contributions to Rocket.Chat given a reaction.
+        """
+        if not self.graded_activity or self.grade == self.weight:
+            return
+        messages = self._get_user_messages(self.user_data["default_group"],
```

Comment:
Only graded on the default group?

Reply:
Yes, because the user can be in a lot of groups